cmd/captplanet/run.go

```diff
@@ -69,6 +72,19 @@ func init() {
 	cfgstruct.Bind(runCmd.Flags(), &runCfg, cfgstruct.ConfDir(defaultConfDir))
 }
 
+func collectErrors(errch chan error, d time.Duration) error {
+	errs := []error{<-errch}
+
+	for {
+		select {
+		case err := <-errch:
+			errs = append(errs, err)
+		case <-time.After(d):
```

Comment:
This restarts the countdown every loop. You can either 1. use a fixed hard deadline or 2. read until empty and then block.

Reply:
That would block because the errch may never return something else.

Reply:
```
errs := []error{<-errch}
timeout := time.After(5 * time.Second)
for {
    select {
    case err := <-errch:
        errs = append(errs, err)
    case <-timeout:
        return errs
    }
}
```

Reply:
👍

Reply:
@egonelbre It can't return errs, since errs is a slice of errors here and it has to return a single error, but otherwise this works perfect. I'll push this up with the other changes.